linkerd/stack/src/proxy.rs

```diff
@@ -19,6 +19,18 @@ pub trait Proxy<Req, S: tower::Service<Self::Request>> {
 
     /// Usually invokes `S::call`, potentially modifying requests or responses.
     fn proxy(&self, inner: &mut S, req: Req) -> Self::Future;
+
+    /// Like [`ServiceExt::oneshot`](tower::util::ServiceExt::oneshot), but with
+    /// a `Proxy` too.
+    fn proxy_oneshot(self, svc: S, req: Req) -> Oneshot<Self, S, Req>
```

Comment:
I'm not necessarily opposed to this but I'm having a little trouble understanding the use case. The whole point of a proxy is to take a service by-reference. It sounds like  what we really want is to fuse the proxy to a service in this case?